templates/documents/upload_files_modal.html

```diff
@@ -0,0 +1,93 @@
+{% extends 'documents/shared/modal_body.html' %}
+
+{% block modal_body %}
+    <div class="mt-2 flex flex-col gap-5">
+        <input
+            x-ref="fileUpload"
+            name="files"
+            @change="handleFileChange($event)"
+            type="file"
+            multiple
+            class="file-input file-input-bordered w-full max-w-xs"
+            accept=".txt,.pdf,.jpg,.png"/>
+
+        <select class="select select-bordered w-full" name="collection_name">
+            {% for collection in collections %}
+                <option value="{{ collection.name }}">{{ collection.name }}</option>
+            {% endfor %}
+        </select>
+
+        <template x-for="(file, index) in uploaded_files" :key="index">
+            <div class="grid grid-cols-2 gap-1 border-b border-base-300 py-3 items-center">
+                <span class="col-span-1" x-text="file.name"></span>
+                <div class="col-span-1 flex flex-row gap-1 items-center">
+                    <textarea
+                        x-model="file_summaries[file.name]"
+                        class="col-span-1 textarea textarea-bordered w-full placeholder:text-gray-500"
+                        maxlength="max_summary_length"
+                        placeholder="Enter a summary of the file..."
+                    ></textarea>
+                    <button aria-label="Remove File" title="Remove File" type="button" class="btn btn-xs" @click="removeFile(index)">
+                        <i class="fa-solid fa-trash htmx-hide"></i>
+                    </button>
+                </div>
+            </div>
+        </template>
+        <input type="hidden" name="file_summaries" :value="JSON.stringify(file_summaries)" />
+    </div>
+{% endblock %}
+
+{% block submit_button %}
+    <button class="btn btn-primary mt-2" :disabled="uploaded_files.length === 0">Submit</button>
+{% endblock %}
+
+{% block extra_js %}
+    <script>
+        document.addEventListener('alpine:init', () => {
+            const megabyte_in_bytes = 1048576;
+            const byte_limit = megabyte_in_bytes * 512;
+            Alpine.data('modalData', () => ({
+                message: "",
+                uploaded_files: [],
+                file_summaries: {},
+                selected_collection: null,
+                total_upload_size_bytes: 0,
```

Comment:
Shouldn't this start at the current collection size? Or is this just to limit to 512MB per batch?

Reply:
Yeah just is just a batch upload limit. I copied this code from the `input_bar.html`. In that context, it's the max limit that assistants can handle. In this context, it doesn't really make much sense to limit the total file size come to think of it, since the limits are on:
1. Individual file size
2. The number of files per collection
3. The sum of all files per team

I haven't really done any of those checks yet, so I'm going to remove this code and create a ticket for the above size checks, otherwise the PR just gets too large.